include/aspect/utilities.h

```diff
@@ -813,6 +813,42 @@ namespace aspect
          */
         std_cxx11::unique_ptr<aspect::Utilities::AsciiDataLookup<1> > lookup;
     };
+
+    /**
+     * A weighted p norm average caluclator:
+     * $\text{weighed\_p\_norm\_average} =
+     * \frac{1}{k}(\sum_{i=1}^k \big(\text{weights}_i \text{values}_i^p)\big)^{\frac{1}{p}}$.
+     *
+     * There have been several special cases implemented, mostly for performance reasons, and
+     * they should not influence the outcome, expect for when n <= -1000 or >= 1000. These are
+     * assumed to be the min and max norm, where it looks at the smallest and largest weight
+     * respectively and returns that value.
+     */
+    double weighted_p_norm_average (const std::vector<double> &weights,
+                                    const std::vector<double> &values,
+                                    const double p);
+
+
+    /**
+     * Computes the derivative of values which have been averaged by a weighted p norm.
+     * If you want to use this for computing the weighted average for the derivatives to
+     * the strain rate and the pressure for the effective viscosity the equation looks like this:
+     * $\frac{\partial\eta_{\text{eff}}}{\partial\dot\varepsilon \lor \partial P} =
```

Comment:
what does `\lor` do?

I think in general that it would would be easier to just write the documentation in terms of functions `f_i(x)` where `f(x)` is some weighted sum of the `f_i`, and the given derivatives are `partial f_i(x)/partial x`. The way you write it you make the assumption (or suggest) that the function is specifically intended for functions `eta(eps)` or `eta(p)`, but this is just a special case and the function is generally applicable.